Add time stamp to logs

jnk5y opened the pull request.

jnk5y commented:
Sorry, but I just noticed the older PR for adding time stamps. When I do a docker logs shepherd I don't get any timestamps and docker doesn't add them so this is really useful for me. I also have code to set the timezone through an env variable if you add timestamps.

djmaze commented:
@jnk5y If you use `docker logs` or `docker service logs`, you just need to add the `-t` parameter to get timestamps as well.

I would prefer not to add functionality which docker itself provides as well.

jnk5y commented:
Docker will only show logs in UTC and with this PR along with some more code I have you can have logs in your preferred timezone. Up to you of course.

djmaze commented:
Not sure. I think I'd rather have this configurable (and disabled by default) via a separate variable.

That'd be the chance to add a `log` function instead of directly calling `echo` everywhere. Would you be willing to change this accordingly?

jnk5y commented:
Yeah, I can tackle that. I can also add a verbose option for those who want all the logs.

djmaze commented:
What exactly do you mean with `all the logs`? That we make the current logging optional?

jnk5y commented:
Minimum logs would be when shepherd updates a service. Verbose would be everything else. If you have 10 services and they are all checked every 5 minutes that's 288*10 logs every day saying a service update was attempted. I really only care to know when an update occurred. I'm happy to make verbose mode the default.

djmaze commented:
That sounds good to me. Go ahead!

jnk5y commented:
I created a logging function and added a verbose logging mode. I also set the timezone from an environment variable. I also moved your environment variables from your dockerfile to the docker-compose file so they are easier to update. I have some merge conflicts that I can't quite fix myself if you could assist.

djmaze commented:
@jnk5y Please leave the environment variables in the Dockerfile as well. They are serving as defaults so people do have to set each of them explicitely (e.g. when running without Docker Compose). Their values can be overridden in the compose file either way.

jnk5y commented:
Done

djmaze commented on shepherd:
```diff
@@ -5,6 +5,15 @@ server_version() {
+  local log="$1"
+  local is_verbose_log="${2:-false}"
+  
+  if [ "$is_verbose_log" == "true" -a "$verbose" == "true" -o "$is_verbose_log" == "false" ]; then
```
I'd rather like to use the (in my opinion better) newer condition syntax with `&&` and `||`.

djmaze commented on shepherd:
```diff
@@ -53,39 +63,44 @@ update_services() {
+  verbose="${VERBOSE:-true}"
+  
+  echo $tz > /etc/timezone
+  logger "Timezone set to $tz"
```
Are you sure this is needed? As the `TZ` variable is set globally for the container AFAIK there is no need to write to the timezone file.

djmaze commented on docker-compose.apprise.yml:
```diff
@@ -14,6 +14,11 @@ services:
       placement:
         constraints:
         - node.role == manager
+    environment:
```
There already is an `environment` section above. Please move the new variables there as well.

jnk5y commented:
Everything should now be resolved. Sorry I'm new at this so not sure if i'm doing things the right way.

djmaze commented:
@jnk5y Thanks! I think we can remove the write to */etc/timezone*, as stated above?

jnk5y commented:
Tested without setting /etc/timezone and it worked so I removed it.
